### ISSUE TITLE
AssertionError: Torch not compiled with CUDA enabled - macOS Sequoia 15.1.1 (intel, AMD) 

### ISSUE THREAD

**Issue by SkyForceCoder** [Tencent/HunyuanVideo#22]
I'm encountering the following error while running the preprocess_text_encoder_tokenizer function in the preprocess_text_encoder_tokenizer_utils.py 

**Environment:**

> **Processor:**

Model: Intel(R) Xeon(R) W-2140B CPU
Speed: 3.20 GHz
Cores: 8 cores (with Hyper-Threading enabled)

> **Memory (RAM):**

Total RAM: 32 GB (34359738368 bytes)

> **GPU:**

Model: Radeon Pro Vega 56
VRAM: 8 GB
Bus: PCIe (x16)
Metal Support: Metal 3 (supports Apple's graphics framework)

**Expected Behaviour**
The model should successfully move to the device without raising the AssertionError. It would be helpful if PyTorch provided support for GPU acceleration on macOS with Intel and AMD GPUs (such as Radeon Pro Vega 56), including integration with Metal for better compatibility on these platforms.

**Actual Behaviour**
AssertionError: Torch not compiled with CUDA enabled

---------------------------------------------------------------------

>python hyvideo/utils/preprocess_text_encoder_tokenizer_utils.py --input_dir ckpts/llava-llama-3-8b-v1_1-transformers --output_dir ckpts/text_encoder

Special tokens have been added in the vocabulary, make sure the associated word embeddings are fine-tuned or trained.
Loading checkpoint shards: 100%|██████████████████████████████████████████████████████████████████████████████████████████████| 4/4 [00:00<00:00,  6.87it/s]
Traceback (most recent call last):
  File "/Volumes/Spidy/GitHub/AI/Video/HunyuanVideo/hyvideo/utils/preprocess_text_encoder_tokenizer_utils.py", line 46, in <module>
    preprocess_text_encoder_tokenizer(args)
  File "/Volumes/Spidy/GitHub/AI/Video/HunyuanVideo/hyvideo/utils/preprocess_text_encoder_tokenizer_utils.py", line 16, in preprocess_text_encoder_tokenizer
    ).to(0)
  File "/Volumes/Spidy/VirtualEnvDir/HunyuanVideo/lib/python3.10/site-packages/transformers/modeling_utils.py", line 2576, in to
    return super().to(*args, **kwargs)
  File "/Volumes/Spidy/VirtualEnvDir/HunyuanVideo/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1160, in to
    return self._apply(convert)
  File "/Volumes/Spidy/VirtualEnvDir/HunyuanVideo/lib/python3.10/site-packages/torch/nn/modules/module.py", line 810, in _apply
    module._apply(fn)
  File "/Volumes/Spidy/VirtualEnvDir/HunyuanVideo/lib/python3.10/site-packages/torch/nn/modules/module.py", line 810, in _apply
    module._apply(fn)
  File "/Volumes/Spidy/VirtualEnvDir/HunyuanVideo/lib/python3.10/site-packages/torch/nn/modules/module.py", line 810, in _apply
    module._apply(fn)
  [Previous line repeated 1 more time]
  File "/Volumes/Spidy/VirtualEnvDir/HunyuanVideo/lib/python3.10/site-packages/torch/nn/modules/module.py", line 833, in _apply
    param_applied = fn(param)
  File "/Volumes/Spidy/VirtualEnvDir/HunyuanVideo/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1158, in convert
    return t.to(device, dtype if t.is_floating_point() or t.is_complex() else None, non_blocking)
  File "/Volumes/Spidy/VirtualEnvDir/HunyuanVideo/lib/python3.10/site-packages/torch/cuda/__init__.py", line 289, in _lazy_init
    raise AssertionError("Torch not compiled with CUDA enabled")
AssertionError: Torch not compiled with CUDA enabled

Created new issue here too
https://github.com/pytorch/pytorch/issues/141973

**Comment by zhoudaquan** [comment]
> I'm encountering the following error while running the preprocess_text_encoder_tokenizer function in the preprocess_text_encoder_tokenizer_utils.py
> 
> **Environment:**
> 
> > **Processor:**
> 
> Model: Intel(R) Xeon(R) W-2140B CPU Speed: 3.20 GHz Cores: 8 cores (with Hyper-Threading enabled)
> 
> > **Memory (RAM):**
> 
> Total RAM: 32 GB (34359738368 bytes)
> 
> > **GPU:**
> 
> Model: Radeon Pro Vega 56 VRAM: 8 GB Bus: PCIe (x16) Metal Support: Metal 3 (supports Apple's graphics framework)
> 
> **Expected Behaviour** The model should successfully move to the device without raising the AssertionError. It would be helpful if PyTorch provided support for GPU acceleration on macOS with Intel and AMD GPUs (such as Radeon Pro Vega 56), including integration with Metal for better compatibility on these platforms.
> 
> **Actual Behaviour** AssertionError: Torch not compiled with CUDA enabled
> 
> > python hyvideo/utils/preprocess_text_encoder_tokenizer_utils.py --input_dir ckpts/llava-llama-3-8b-v1_1-transformers --output_dir ckpts/text_encoder
> 
> Special tokens have been added in the vocabulary, make sure the associated word embeddings are fine-tuned or trained. Loading checkpoint shards: 100%|██████████████████████████████████████████████████████████████████████████████████████████████| 4/4 [00:00<00:00, 6.87it/s] Traceback (most recent call last): File "/Volumes/Spidy/GitHub/AI/Video/HunyuanVideo/hyvideo/utils/preprocess_text_encoder_tokenizer_utils.py", line 46, in preprocess_text_encoder_tokenizer(args) File "/Volumes/Spidy/GitHub/AI/Video/HunyuanVideo/hyvideo/utils/preprocess_text_encoder_tokenizer_utils.py", line 16, in preprocess_text_encoder_tokenizer ).to(0) File "/Volumes/Spidy/VirtualEnvDir/HunyuanVideo/lib/python3.10/site-packages/transformers/modeling_utils.py", line 2576, in to return super().to(*args, **kwargs) File "/Volumes/Spidy/VirtualEnvDir/HunyuanVideo/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1160, in to return self._apply(convert) File "/Volumes/Spidy/VirtualEnvDir/HunyuanVideo/lib/python3.10/site-packages/torch/nn/modules/module.py", line 810, in _apply module._apply(fn) File "/Volumes/Spidy/VirtualEnvDir/HunyuanVideo/lib/python3.10/site-packages/torch/nn/modules/module.py", line 810, in _apply module._apply(fn) File "/Volumes/Spidy/VirtualEnvDir/HunyuanVideo/lib/python3.10/site-packages/torch/nn/modules/module.py", line 810, in _apply module._apply(fn) [Previous line repeated 1 more time] File "/Volumes/Spidy/VirtualEnvDir/HunyuanVideo/lib/python3.10/site-packages/torch/nn/modules/module.py", line 833, in _apply param_applied = fn(param) File "/Volumes/Spidy/VirtualEnvDir/HunyuanVideo/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1158, in convert return t.to(device, dtype if t.is_floating_point() or t.is_complex() else None, non_blocking) File "/Volumes/Spidy/VirtualEnvDir/HunyuanVideo/lib/python3.10/site-packages/torch/cuda/**init**.py", line 289, in _lazy_init raise AssertionError("Torch not compiled with CUDA enabled") AssertionError: Torch not compiled with CUDA enabled
> 
> Created new issue here too [pytorch/pytorch#141973](https://github.com/pytorch/pytorch/issues/141973)

Thank you for raising this. We will consider the optimization on different types of GPUs/encironments. However, we have a list of TODOs currently,,, I cannot guarantee the time span for this.... in the meanwhile, you are more than welcome to help contribute to this!

Best regards,
DQ

**Comment by SkyForceCoder** [comment]
Hi @zhoudaquan 

After testing the environment, we’ve confirmed that GPU acceleration is available via PyTorch’s MPS backend on macOS. Here’s the summary of our setup:


% conda activate HunyuanVideo
% python --version
Python 3.10.9
% python -c "import torch; print(torch.__version__)"
2.1.1
% python -c "import torch; print(torch.backends.mps.is_available())"
True
% pip show torch

Name: torch
Version: 2.1.1
Summary: Tensors and Dynamic neural networks in Python with strong GPU acceleration
Home-page: https://pytorch.org/
Author: PyTorch Team
Author-email: packages@pytorch.org
License: BSD-3
Location: /Volumes/Spidy/VirtualEnvDir/HunyuanVideo/lib/python3.10/site-packages
Requires: filelock, fsspec, jinja2, networkx, sympy, typing-extensions
Required-by: accelerate, torchvision

This indicates that GPU acceleration can now be leveraged on Intel and AMD macOS systems with Metal support. We request you to implement support for the MPS backend in the project so macOS users, especially those using Intel and AMD hardware, can benefit from significant performance improvements.

This enhancement will broaden accessibility and optimize performance for a wider audience.

Thank you for considering this!

**Comment by BoBo0037** [comment]
you can check my repo for mac.  https://github.com/BoBo0037/bobo-fun .now, i cannot run within bfloat16...

**Comment by SkyForceCoder** [comment]
Thanks @BoBo0037I created a new issue there regarding the problem with Intel Macs. I’ve included detailed instructions for reference. I will create a new issue after resolving these issues.

**Comment by Dhrhciebcy** [comment]
I simply resolved this problem uninstalling the previous version of Torch and installing Torch 2.1.1 with CUDA

`python -m pip uninstall torch`
`python -m pip install torch==2.1.1 --index-url https://download.pytorch.org/whl/cu121`

**Comment by SkyForceCoder** [comment]
@Dhrhciebcy but it's not for intel amd MAC, on which machine your are running it I think nvidia

% python -m pip install torch==2.1.1 --index-url https://download.pytorch.org/whl/cu121
Looking in indexes: https://download.pytorch.org/whl/cu121
ERROR: Could not find a version that satisfies the requirement torch==2.1.1 (from versions: none)
ERROR: No matching distribution found for torch==2.1.1

**Comment by Dhrhciebcy** [comment]
> @Dhrhciebcy but it's not for intel amd MAC, on which machine your are running it I think nvidia
> 
> % python -m pip install torch==2.1.1 --index-url https://download.pytorch.org/whl/cu121 Looking in indexes: https://download.pytorch.org/whl/cu121 ERROR: Could not find a version that satisfies the requirement torch==2.1.1 (from versions: none) ERROR: No matching distribution found for torch==2.1.1

I'm sorry, you're right, I have an NVIDIA graphics card, I didn't realize that the graphics card was AMD. I'm stupid

**Comment by SkyForceCoder** [comment]
@Dhrhciebcy  https://github.com/Tencent/HunyuanVideo/issues/22#issue-2715529819
